plugins/CoreHome/Columns/Metrics/EvolutionMetric.php

```diff
@@ -170,4 +190,56 @@ private function getPastDataTable()
         }
         return $result;
     }
+
+    /**
+     * Calculate the ratio of time between a past period and current incomplete period
+     *
+     * eg. if today is Thursday at 12:00pm and the past period is a week then the ratio is 0.5, exactly half of the
+     * current incomplete period has passed
+     *
+     * If the current period end is in the past then the ratio will always be 1, since the current period is complete.
+     *
+     * @param DataTable|null $currentData
+     * @param DataTable|null $pastData
+     * @param Row $row
+     * @return float|int
+     * @throws \Exception
+     */
+    public static function getRatio(?DataTable $currentData, ?DataTable $pastData, Row $row)
+    {
+        $ratio = 1;
+
+        if ($currentData != null && $pastData != null) {
+
+            $p = $pastData->getMetadata(DataTableFactory::TABLE_METADATA_PERIOD_INDEX);
+
+            $pStart = $p->getDateStart()->setTime('00:00:00');
+            $pEnd = $p->getDateEnd()->setTime('00:00:00');
```

Comment:
This needs to be the end of the period, otherwise the last day won't count. That's also the reason why this doesn't work for a day period at the moment at all.
```suggestion
            $pEnd = $p->getDateEnd()->setTime('23:59:59');
```

Reply:
:facepalm: Good spot, thanks @sgiehl